galley/pkg/config/analysis/analyzers/auth/servicerolebindings_test.go

```diff
@@ -0,0 +1,40 @@
+package auth
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	"istio.io/api/rbac/v1alpha1"
+	"istio.io/istio/galley/pkg/config/processor/metadata"
+	"istio.io/istio/galley/pkg/config/resource"
+	"istio.io/istio/galley/pkg/config/testing/fixtures"
+)
+
+func TestWithMatch(t *testing.T) {
+	g := NewGomegaWithT(t)
+
+	ctx := fixtures.NewContext()
+	ctx.AddEntry(metadata.IstioRbacV1Alpha1Servicerolebindings, resource.NewName("default", "srb"), &v1alpha1.ServiceRoleBinding{
+		RoleRef: &v1alpha1.RoleRef{Name: "service_role"},
+	})
+	ctx.AddEntry(metadata.IstioRbacV1Alpha1Serviceroles, resource.NewName("default", "service_role"), &v1alpha1.ServiceRole{})
+
+	(&ServiceRoleBindingAnalyzer{}).Analyze(ctx)
+	g.Expect(ctx.Reports).To(BeEmpty())
+}
+
+func TestNoMatch(t *testing.T) {
+	g := NewGomegaWithT(t)
+
+	ctx := fixtures.NewContext()
+	srbName := resource.NewName("default", "srb")
+	ctx.AddEntry(metadata.IstioRbacV1Alpha1Servicerolebindings, resource.NewName("default", "srb"), &v1alpha1.ServiceRoleBinding{
+		RoleRef: &v1alpha1.RoleRef{Name: "bogus_service_role"},
+	})
+
+	(&ServiceRoleBindingAnalyzer{}).Analyze(ctx)
+	srbReports := ctx.Reports[metadata.IstioRbacV1Alpha1Servicerolebindings]
+	g.Expect(srbReports).To(HaveLen(1))
+	g.Expect(srbReports[0].Origin).To(BeEquivalentTo(srbName.String()))
```

Comment:
The main thing we should check for is the error code in most of the cases.  (In addition to  Checking the origin that is.

In fact, I'd recommend you to start thinking about how to create a testing model for these analyzers in general. I'd like us to provide the facilities for writing & testing analyzers and allow other Istio developers to write the analyzers, for the most part. One aspect of it is the ability to quickly add tests.

For example, we can create a folder based approach, where for a given set of input files (in yaml form), we can have an expected output file, baselining the error messages/details. This would make adding new tests for an analyzer easy.

Reply:
I'm thinking along very similar lines, and that's about where I wanted to iterate next.